laravel/app/EvaluationParser.php

```diff
@@ -14,248 +8,441 @@
+                                $diagnosis = $line[1];
+                                $procedure = $line[2];
+                                $location = $line[3];
+                                $asa = $line[4];
+
+                                $rId = $resident['id'];
+                                $rName = $resident['dbName'];
+                                $aId = $attending['id'];
+                                $aName = $attending['dbName'];
+                                
+                                //echo "\nEvaluation Data: " . $dateInDB . ": Diagnosis(" . $diagnosis . ") Resident(" . $rName . ") Attending(" . $aName . ") Diff(". $minutesOverlapped . ")";
+                                if (EvaluateData::where(['date' => $dateInDB, 'location' => $location, 'resident' => $rName, 'attending' => $aName])->doesntExist()) {
+                                    Log::info("Adding Evaluation Data: " . $dateInDB . ": Diagnosis(" . $diagnosis . ") Resident(" . $rName . ") Attending(" . $aName . ") Diff(". $minutesOverlapped . ")");
+                                    EvaluateData::insert([
+                                        'date' => $dateInDB, 
+                                        'location' => $location, 
+                                        'diagnosis' => $diagnosis, 
+                                        'procedure' => $procedure, 
+                                        'asa' => $asa, 
+                                        'rId' => $rId, 
+                                        'resident' => $rName, 
+                                        'aId' => $aId, 
+                                        'attending' => $aName, 
+                                        'diff' => $minutesOverlapped,
+                                        'created_at'=>Carbon::now()
+                                    ]);
+                                }
+                            }
+                        }
+                    }
+                }
             }
+        }
+        //send emails for all users not added 
+        self::notifyForAllFailedUsers($this->residentsFailedAdding, $this->attendingsFailedAdding, "Melanie", "ferguson.881@osu.edu");
+        fclose($fp);
+    }
 
-            $namefl = $name_first." ".$name_last;
-
+    private function getParticipants($line) {
+        $residentsInLine = array();
+        $attendingsInLine = array();
+        $line = $line . "\n";
+        // while there are still participants in line, parse and add to residents or attendings
+        while ($line != false) {
+            // get participant's full name
+            $name = substr($line, 0, stripos(substr($line, 0), ","));
             $ep = stripos(substr($line, 0), "\n");
-            $line=substr($line,$ep+1);
-
-
-            while(substr($line,0,1)==" "){
+            $line = substr($line, $ep + 1);
+            // parse participant information
+            while (substr($line, 0, 1) == " ") {
```

Comment:
isn't `substr($line, 0, 1)` the same as `$line[0]`?

PHP may even have a method to check if a string starts with another string, which might be even easier to read.